code/modules/supermatter/supermatter.dm

```diff
@@ -390,12 +386,16 @@
 
 	playsound(get_turf(src), 'sound/effects/supermatter.ogg', 50, 1)
 
+	if(istype(W, /obj/item/weapon/holder))
+		var/obj/item/weapon/holder/H = W
+		if(H.stored_mob)
+			Consume(H.stored_mob)
+
```

Comment:
This whole thing is a bit too snowflakey for me.  The holder is the one that has the odd behaviour, and it should be the one to perform whatever steps are needed to correct any weird shit that happens when it's deleted.  For example, moving this logic to Destroy().